src/resource/page/ResourcePage.tsx

```diff
@@ -142,12 +143,22 @@ export default function ResourcePage({ id }: { id: string }): JSX.Element {
       .getResource(id)
       .then(async (r: Resource) => {
         setResource(r);
-        if (r.children.length > 0) {
-          // fetch children
-          const childR = await api
-            .getChildResourcesByParentId(r.id)
-            .then((resourcesResponse) => resourcesResponse.results);
-          setChildResources(childR);
+        const resourceHasChildren = r.children.length > 0;
+        const resourceHasParents = r.parents.length > 0;
+
+        if (resourceHasChildren || resourceHasParents) {
+          // fetch children and parents
+          const [childsResponse, parentsResponse] = await Promise.all([
+            resourceHasChildren
+              ? api.getChildResourcesByParentId(r.id)
+              : { results: [] },
```

Comment:
I think this might be a bit simpler if the api-method would return a list?

Reply:
- [ ] Return list from api-method